internal/fileutils/fileutils_test.go

```diff
@@ -0,0 +1,145 @@
+package fileutils_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/fileutils"
+)
+
+func TestFileExists(t *testing.T) {
+	tests := []struct {
+		name            string
+		fileExists      bool
+		parentDirIsFile bool
+
+		expectExists bool
+		expectError  bool
+	}{
+		{name: "Returns true when file exists", fileExists: true, expectExists: true},
+		{name: "Returns false when file does not exist", fileExists: false, expectExists: false},
+		{name: "Returns false when parent directory does not exist", fileExists: false, expectExists: false},
+
+		{name: "Error when parent directory is a file", parentDirIsFile: true, expectError: true},
+	}
```

Comment:
I'm a bit confused about this... Why did you use a `[]struct{}` instead of our usual `map[string]struct{}`?

Reply:
fixed

Reply:
I fixed it one of the tests in this file but forgot the other ones. Fixed it in all of them now.